cmd/bbolt/command_check.go

```diff
@@ -0,0 +1,77 @@
+package main
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/spf13/cobra"
+
+	bolt "go.etcd.io/bbolt"
+	"go.etcd.io/bbolt/internal/guts_cli"
+)
+
+func newCheckCobraCommand() *cobra.Command {
+	checkCmd := &cobra.Command{
+		Use:   "check",
+		Short: "verifies integrity of bbolt database",
+		Long: strings.TrimLeft(`
+usage: bolt check PATH
+
+Check opens a database at PATH and runs an exhaustive check to verify that
+all pages are accessible or are marked as freed. It also verifies that no
+pages are double referenced.
+
+Verification errors will stream out as they are found and the process will
+return after all pages have been checked.
+`, "\n"),
+		Args: func(cmd *cobra.Command, args []string) error {
+			if len(args) == 0 || args[0] == "" {
+				return ErrPathRequired
+			}
+			if len(args) > 1 {
+				return ErrTooManyArgs
+			}
+			return nil
```

Comment:
This may actually be a question for @ahrtr. Is there any particular reason why we're not using https://pkg.go.dev/github.com/spf13/cobra#ExactArgs. I know that the `surgery` sub-commands do the check this same way. Is it to give more context to the error? i.e., rather than erroring with a wrong number of arguments, saying that the first argument needs to be a path?

Reply:
I think it's mostly a copy&paste thing :) We totally should use ExactArgs.

Reply:
> https://pkg.go.dev/github.com/spf13/cobra#ExactArgs.

Yes, I agree that `ExactArgs` can simplify the implementation.  It's a generic problem, so it's OK to fix it separately.

Reply:
> Is it to give more context to the error? i.e., rather than erroring with a wrong number of arguments, saying that the first argument needs to be a path?

Once an error is returned, I think cobra will automatically print the usage, in which all info is included. So it seems that we don't need to provide any extra context.

Reply:
raised https://github.com/etcd-io/bbolt/pull/728